app/helpers/application_helper/button/middleware_domain_server_action.rb

```diff
@@ -1,12 +1,5 @@
 class ApplicationHelper::Button::MiddlewareDomainServerAction < ApplicationHelper::Button::MiddlewareServerAction
-
   def visible?
-    in_domain? && super
-  end
-
-  private
-
-  def in_domain?
-    @record.present? && (@record.try(:in_domain?) || @record.try(:middleware_server).try(:in_domain?))
+    @record.try(:in_domain?) && super
```

Comment:
please explain why is this logic reduce ok (in other words - how does that maintain the functionality)
thanks